Doesn't compile on mac os x 64 bit

Any plans to make it compile on mac?

c++ -fPIC -D__LITTLE_ENDIAN__    -c -o sfklCoding.o sfklCoding.cpp
sfklCoding.cpp:551:8: warning: comparison of unsigned expression < 0 is always false [-Wtautological-compare]
        if (n < 0  ||  n > ZBUF_SIZE)   // Check for valid block length
            ~ ^ ~
sfklCoding.cpp:553:78: warning: format specifies type 'long' but the argument has type 'uint32_t' (aka 'unsigned int') [-Wformat]
          sprintf(MsgTxt, "ERROR - Invalid length for Non-audio Block (apparently %ld bytes) %s", n, CorruptedMsg);
                                                                                  ~~^             ~
                                                                                  %d
2 warnings generated.
c++ -fPIC -D__LITTLE_ENDIAN__    -c -o sfklDiff.o sfklDiff.cpp
c++ -fPIC -D__LITTLE_ENDIAN__    -c -o sfklLPC.o sfklLPC.cpp
c++ -fPIC -D__LITTLE_ENDIAN__    -c -o sfklZip.o sfklZip.cpp
c++ -fPIC -D__LITTLE_ENDIAN__    -c -o sfklCrunch.o sfklCrunch.cpp
c++ -fPIC -D__LITTLE_ENDIAN__    -c -o sfklFile.o sfklFile.cpp
c++ -fPIC -D__LITTLE_ENDIAN__    -c -o sfklString.o sfklString.cpp
c++ -shared sfklCoding.o sfklDiff.o sfklLPC.o sfklZip.o sfklCrunch.o sfklFile.o sfklString.o -o libsfark.so
Undefined symbols for architecture x86_64:
  "sfkl_DisplayNotes(char const_, char const_)", referenced from:
      ExtractTextFile(BLOCK_DATA_, unsigned long) in sfklCoding.o
  "sfkl_UpdateProgress(int)", referenced from:
      sfkl_Decode(char const_, char const_) in sfklCoding.o
  "sfkl_GetLicenseAgreement(char const_, char const_)", referenced from:
      ExtractTextFile(BLOCK_DATA_, unsigned long) in sfklCoding.o
  "sfkl_msg(char const_, int)", referenced from:
      ReadHeader(V2_FILEHEADER_, unsigned char_, int) in sfklCoding.o
      InvalidEncodeCount(int, int) in sfklCoding.o
      DecompressTurbo(BLOCK_DATA_, unsigned short) in sfklCoding.o
      CheckShift(short_, unsigned short, short_, short_) in sfklCoding.o
      DecompressFast(BLOCK_DATA_, unsigned short) in sfklCoding.o
      ProcessNextBlock(BLOCK_DATA_) in sfklCoding.o
      ExtractTextFile(BLOCK_DATA_, unsigned long) in sfklCoding.o
      ...
  "_adler32", referenced from:
      ReadHeader(V2_FILEHEADER_, unsigned char_, int) in sfklCoding.o
      ProcessNextBlock(BLOCK_DATA_) in sfklCoding.o
      ExtractTextFile(BLOCK_DATA_, unsigned long) in sfklCoding.o
  "_uncompress", referenced from:
      UnMemcomp(unsigned char const_, int, unsigned char_, int) in sfklZip.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: **\* [libsfark.so] Error 1

--- a/Makefile
+++ b/Makefile
@@ -11,9 +11,10 @@ OS := $(shell uname)
 ifeq ($(OS),Darwin)
 LDFLAGS += -flat_namespace -undefined suppress -dynamiclib
 SO = dylib
-else
+else 
 LDFLAGS += -shared
 SO = so
+INSTALL += -D
 endif
 
 all: libsfark.$(SO)
@@ -25,5 +26,5 @@ libsfark.$(SO): $(OBJECTS)
 	$(CXX) -shared $(LDFLAGS) $(OBJECTS) -o libsfark.$(SO)
 
 install: libsfark.$(SO) sfArkLib.h
-	$(INSTALL) -D libsfark.$(SO) $(DESTDIR)/usr/local/lib/libsfark.$(SO)
-	$(INSTALL) -D sfArkLib.h $(DESTDIR)/usr/local/include/sfArkLib.h
+	$(INSTALL) libsfark.$(SO) $(DESTDIR)/usr/local/lib/libsfark.$(SO)
+	$(INSTALL) sfArkLib.h $(DESTDIR)/usr/local/include/sfArkLib.h

